python/tests/test_tables.py

```diff
@@ -1913,6 +1947,13 @@ def test_from_dict(self):
         t2 = tskit.TableCollection.fromdict(d)
         self.assertEquals(t1, t2)
 
+    def test_roundtrip_dict(self):
+        ts = msprime.simulate(10, mutation_rate=1, random_seed=1)
+        t1 = ts.tables
+        self.add_metadata(t1)
+        t2 = tskit.TableCollection.fromdict(t1.asdict())
+        self.assertEqual(t1, t2)
```

Comment:
Can we add another test here to make sure that we also round-trip without adding the metadata?

Reply:
https://github.com/tskit-dev/tskit/pull/687/commits/57edc3d76982290667c5039ff42cf4f88743e7db